partiql-types/src/main/kotlin/org/partiql/types/StaticType.kt

```diff
@@ -360,13 +364,70 @@ public data class TimeType(
     }
 }
 
-public data class TimestampType(override val metas: Map<String, Any> = mapOf()) : SingleType() {
-    override val allTypes: List<StaticType>
-        get() = listOf(this)
+public class TimestampType internal constructor(internal val timestampType: PartiQLTimestampType = PartiQLTimestampType()) : SingleType() {
+
+    override val metas: Map<String, Any> = timestampType.metas
+    override val allTypes: List<StaticType> = listOf(this)
+    @PartiQLTimestampExperimental
+    public val precision: Int? = timestampType.precision
+    @PartiQLTimestampExperimental
+    public val withTimeZone: Boolean = timestampType.withTimeZone
+
+    // Preserve the original semantics. An arbitrary timestamp type without time zone
+    public constructor(metas: Map<String, Any> = mapOf()) : this(PartiQLTimestampType(null, true, metas))
```

Comment:
```suggestion
    public constructor(metas: Map<String, Any> = mapOf()) : this(PartiQLTimestampType(null, false, metas))
```

Reply:
Actually. The comment is wrong. Ion Timestamp always has time zone ( unknown or UTC-offset).

Reply:
I will update the comment.